execute/internal/gas/evm/gas_helpers.go

```diff
@@ -54,7 +92,28 @@ func (gp EstimateProvider) CalculateMessageMaxGas(msg ccipocr3.Message) uint64 {
 	var data []byte = msg.Data
 	dataLength := len(data)
 
-	// TODO: parse max gas from ExtraArgs.
+	// If msg is an EVM message, its ExtraArgs field is an encoded EVMExtraArgsV2 struct, defined in Client.sol here:
```

Comment:
should reference the contracts in chainlink repo, afaik v2 code has been ported and actively developed there

Reply:
I think it's okay if it's a permanent link. contracts shouldn't be changing anyways.

Reply:
I think it's better to have a link that may one day break than not have any link